packages/stargate/src/stargateclient.ts

```diff
@@ -22,6 +23,45 @@ export interface SequenceResponse {
   readonly sequence: number;
 }
 
+export interface BroadcastTxFailure {
+  readonly height?: number;
+  readonly code: number;
+  readonly transactionHash: string;
+  readonly rawLog?: string;
+  readonly data?: Uint8Array;
+}
+
+export interface BroadcastTxSuccess {
+  readonly height?: number;
```

Comment:
Can height be unset in the success case?

Reply:
I was forced to add this by the compiler, so if it can't we need to update the Tendermint type.

Reply:
Oh, we only have one TM type for success and failure.

Reply:
We split the success/error case depending on `code` in `StargateClient`. I think this is a good place to `assert(typeof response.height === "number")`. If this throws, something unexpected happems that needs debugging.

Reply:
It seems like the height is present for both cases, and set to 0 for failure.

Reply:
Yeah, well, you never know if 0 is omitted in the result from a Go interface